### PR TITLE
OCPBUGS-23801: update OTEL `google.golang.org/grpc/otelgrpc`

### PR DESCRIPTION
**Pull request by rexagod**
Refer: https://github.com/open-telemetry/opentelemetry-go-contrib/security/advisories/GHSA-8pgv-569h-w5rw.

***

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -103,31 +103,31 @@ require (
 	go.etcd.io/etcd/api/v3 v3.5.10 // indirect
 	go.etcd.io/etcd/client/pkg/v3 v3.5.10 // indirect
 	go.etcd.io/etcd/client/v3 v3.5.10 // indirect
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.42.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.50.0 // indirect
```
Comment (marioferh) [inline go.mod:106]: fix included in v0.46.0

**Review by marioferh — COMMENTED** [review]
/lgtm

**Comment by rexagod** [comment]
Rebased, please /lgtm again.

**Pull request merged**